Add docs for `vars.yml` (#8705)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/var.md

description: "Pass variables from `dbt_project.yml` file into models."
---

<VersionBlock lastVersion="1.11">

Variables can be passed from your [`dbt_project.yml`](/reference/dbt_project.yml) file into models during compilation. These variables allow you to make your models configurable instead of hardcoding values directly in SQL. For example, you might define a default reporting date, region, or feature flag once in your project and reference it across multiple models.

Variables defined in your `dbt_project.yml` file act as project-wide defaults. You can override them at runtime using the `--vars` command-line argument. For example, to test a different date range or run models with environment-specific settings without modifying your model logic.

</VersionBlock>

<VersionBlock firstVersion="1.12">

> You can pass down variables from your `vars.yml` or `dbt_project.yml` file into models during compilation. These variables are useful for configuring packages for deployment in multiple environments, or defining values that should be used across multiple models within a package.

Reviewer: ```suggestion
You can provide variables from `vars.yml` or `dbt_project.yml` to models during compilation. These variables are useful for configuring packages for deployment in multiple environments, or for defining values that should be used across multiple models within a package.
```

> Variables defined in your `vars.yml` or `dbt_project.yml` file act as project-wide defaults. You can override them at runtime using the `--vars` command-line argument. For example, to test a different date range or run models with environment-specific settings without modifying your model logic.

Reviewer: ```suggestion
Variables defined in `vars.yml` or `dbt_project.yml` act as project-wide defaults. You can override them at runtime using the `--vars` command-line argument. For example, when testing with a different date range or running models with environment-specific settings without changing your model logic.
```

</File>

</VersionBlock>

<VersionBlock firstVersion="1.12">

To define a variable in your project, add the `vars:` config to a dedicated `vars.yml` file or to your `dbt_project.yml` file. `vars.yml` is parsed _before_ `dbt_project.yml`, so you can reference variables from `vars.yml` in `dbt_project.yml` using `{{ var('...') }}`.

<Tabs>
<TabItem value="vars.yml" label="vars.yml">

<File name='vars.yml'>

```yaml
vars:
  event_type: activation
```

</File>
:::info Beta feature

> The `vars.yml` file is a beta feature in <Constant name="core" /> v1.12.

Reviewer: Hiya @luna-bianca 

Should this callout be a snippet?

Author: There is an outstanding query [here](https://github.com/dbt-labs/docs.getdbt.com/pull/8705/changes#r2890652246) about this callout. Once we finalize the language used here I can convert this to a snippet.
